src/components/Footer/Footer.js

```diff
@@ -26,8 +39,18 @@ const Footer = ({
   blockStatus,
   loadingVote,
   showScoreButton = false,
-  onPressScore
+  onPressScore,
+  isShowDM = false
 }) => {
+  const {sendMessageDM} = useDMMessage();
+  const [profile] = React.useContext(Context).profile;
+  const [loading, setLoading] = React.useState({
+    loadingDm: false,
+    loadingDmAnon: false
+  });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

New state variables and hooks are introduced here. It's important to handle potential errors and edge cases that might arise from these new additions. For instance, what happens if `useDMMessage` or `useCreateChat` fail? Consider adding error handling logic to these hooks.
```javascript
const { sendMessageDM, error: dmError } = useDMMessage();
const { createSignChat, error: chatError } = useCreateChat();

// handle errors...
if (dmError) {
  // handle error...
}

if (chatError) {
  // handle error...
}
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->